### PR TITLE
Support new Qiskit release notes

### LINKED ISSUE #281: Add historical release notes for Qiskit API
For Qiskit, it historically has a single release note file for everything: https://qiskit.org/documentation/release_notes.html. But we probably want to migrate to one file per version. It's way too big otherwise.

Where should the release note live in the UI? 

Once we have a design, we need to:

1. Port the original release note file from RST to MDX. This will take forever..I'm not sure how we want to pull it off. We may want to instead not include a ton of historical release notes? Point to the RST file?
2. Figure out how Qiskit will do release notes in the future, if they keep using Reno. It's been working well for them and I suspect they'll want to keep using Reno. But how do we integrate that into MDX?

### LINKED ISSUE #354: Qiskit release notes should have an index page
We realized it's important that `/api/qiskit/release-notes` works and you don't have to go to a dedicated page like `/release-notes/0.45`. That is useful for our redirect logic (https://github.com/Qiskit/documentation/issues/216) and in communication like blogs for us to have a link that always works.

The page should simply be a list of each of the versions we have and link to them. Like duplicating the left sidebar. 

--

Maybe awkwardly, this will be an orphan page, meaning you can only access it from the URL, and not the left side bar. This doesn't work in `_toc.json`:

```json
      "title": "Release Notes",
      "url": "/api/qiskit/release-notes/index",
      "children": [
       ]
```

Abby M and I agreed that's fine. It's still useful to have the stable URL `/release-notes`. We want to be careful to not add meaningful content to the page though, since it's not very discoverable.

### PR DESCRIPTION
**Pull request by frankharkins**
Continuation of #346.

---

Closes #281, closes #354

### CODE REVIEW

File: docs/api/qiskit-ibm-provider/release-notes.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ description: Changes made to Qiskit IBM Provider
 in_page_toc_max_heading_level: 2
 ---
 
-# Release Notes
+<span id="release-notes" />
```
Comment (frankharkins) [inline docs/api/qiskit-ibm-provider/release-notes.md:7]: Not sure why this is but it seems harmless

File: scripts/commands/updateApiDocs.ts [hunk 2]
```diff
@@ -323,6 +343,21 @@ async function convertHtmlToMarkdown(
     JSON.stringify(toc, null, 2) + "\n",
   );
 
+  if (pkg.hasSeparateReleaseNotes) {
```
Comment (Eric-Arellano) [inline scripts/commands/updateApiDocs.ts:346]: There's now a lot of release notes logic in this file and the file is already pretty large. Thoughts on creating a new `lib/releaseNotes.ts` file?

File: scripts/lib/sphinx/sphinxHtmlToMarkdown.ts [hunk 3]
```diff
@@ -64,6 +66,10 @@ export async function sphinxHtmlToMarkdown(options: {
     }
   });
 
+  if (isReleaseNotes && releaseNotesTitle) {
+    $page("h1").html(releaseNotesTitle);
+  }
```
Comment (Eric-Arellano) [inline scripts/lib/sphinx/sphinxHtmlToMarkdown.ts:71]: What does this do?

**Review by Eric-Arellano — APPROVED** [review]
💜

File: scripts/commands/updateApiDocs.ts (hunk 2 above)
Comment (frankharkins) [reply scripts/commands/updateApiDocs.ts:346]: That's a good idea, I'll merge but an issue for it after summit.

File: scripts/lib/sphinx/sphinxHtmlToMarkdown.ts (hunk 3 above)
Comment (frankharkins) [reply scripts/lib/sphinx/sphinxHtmlToMarkdown.ts:71]: This replaces the title "Release Notes" with something like "Qiskit 0.45 release notes", I'll add a comment.

**Pull request merged**